core/data/event.go

```diff
@@ -569,23 +540,18 @@ func getEventByDeviceHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// Get the device
-	deviceFound := true
-	// Try by ID
-	d, err := mdc.Device(deviceId)
+	d, err := mdc.CheckForDevice(deviceId)
```

Comment:
Only perform this check if MetaDataCheck is enabled. There are more cases like this one.